cmd/dockerd/daemon.go

```diff
@@ -515,6 +518,29 @@ func initRouter(opts routerOptions) {
 			grpcBackends = append(grpcBackends, b)
 		}
 	}
+
+	plugins := daemonplugin.Graph()
+	ctx := context.Background()
+	initialized := daemonplugin.NewPluginSet()
+	grpcPlugins := []*daemonplugin.Plugin{}
+	for _, p := range plugins {
+		cx := daemonplugin.NewPluginContext(ctx, p, initialized, opts.daemon)
+		res := p.Init(cx)
+		if err := initialized.Add(res); err != nil {
+			return errors.Wrap(err, "could not add plugin")
+		}
+		grpcPlugins = append(grpcPlugins, res)
```

Comment:
Why the grpcPlugins slice here? (looks like it's only used below to loop over it again, so both loops could be combined?)

Reply:
Fixed